modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpansionModuleModel.java

```diff
@@ -160,6 +253,26 @@ private ArtifactAst expand(List<ComponentAst> moduleComponentModels, Set<String>
                            comp -> false);
   }
 
+  private boolean existOperationThatUsesImplicitConfiguration() {
+    return applicationModel.topLevelComponentsStream()
+        .filter(componentAst -> componentAst.getIdentifier().getName().equals("flow")
+            || componentAst.getIdentifier().getName().equals("sub-flow"))
+        .flatMap(ComponentAst::directChildrenStream)
```

Comment:
Does this only get the first level of its children? What if the operation is in a scope like a for-each, asyn or try scopes.

Reply:
☝🏼

Reply:
there are some recursive iterate methods un `MuleAstUtils`

Reply:
I didn't find any that suit me.